avm/res/sql/server/database/main.bicep

```diff
@@ -2,146 +2,148 @@ metadata name = 'SQL Server Database'
 
-@description('Optional. The resource ID of the elastic pool containing this database.')
-param elasticPoolId string = ''
+@description('Optional. The resource identifier of the restorable dropped database associated with create operation of this database.')
+param restorableDroppedDatabaseId string?
 
-@description('Optional. Location for all resources.')
-param location string = resourceGroup().location
+@description('Optional. Point in time (ISO8601 format) of the source database to restore when createMode set to Restore or PointInTimeRestore.')
+param restorePointInTime string?
 
-@description('Optional. The diagnostic settings of the service.')
-param diagnosticSettings diagnosticSettingType
+@description('Optional. The name of the sample schema to apply when creating this database.')
+param sampleName string = ''
 
-@description('Optional. Specifies the mode of database creation.')
-@allowed([
-  'Default'
-  'Copy'
-  'OnlineSecondary'
-  'PointInTimeRestore'
-  'Recovery'
-  'Restore'
-  'RestoreLongTermRetentionBackup'
-  'Secondary'
-])
-param createMode string = 'Default'
-
-@description('Optional. Resource ID of database if createMode set to Copy, Secondary, PointInTimeRestore, Recovery or Restore.')
-param sourceDatabaseResourceId string = ''
+@description('Optional. The secondary type of the database if it is a secondary.')
+param secondaryType 'Geo' | 'Named' | 'Standby'?
 
 @description('Optional. The time that the database was deleted when restoring a deleted database.')
-param sourceDatabaseDeletionDate string = ''
+param sourceDatabaseDeletionDate string?
 
-@description('Optional. Resource ID of backup if createMode set to RestoreLongTermRetentionBackup.')
-param recoveryServicesRecoveryPointResourceId string = ''
+@description('Optional. The resource identifier of the source database associated with create operation of this database.')
+param sourceDatabaseId string?
 
-@description('Optional. Point in time (ISO8601 format) of the source database to restore when createMode set to Restore or PointInTimeRestore.')
-param restorePointInTime string = ''
+@description('Optional. The resource identifier of the source associated with the create operation of this database.')
+param sourceResourceId string?
 
-@description('Optional. The storage account type to be used to store backups for this database.')
-@allowed([
-  'Geo'
-  'Local'
-  'Zone'
-  ''
-])
-param requestedBackupStorageRedundancy string = ''
+@description('Optional. Whether or not the database uses free monthly limits. Allowed on one database in a subscription.')
+param useFreeLimit bool?
 
-@description('Optional. Whether or not this database is a ledger database, which means all tables in the database are ledger tables. Note: the value of this property cannot be changed after the database has been created.')
-param isLedgerOn bool = false
+@description('Optional. Whether or not this database is zone redundant.')
+param zoneRedundant bool?
```

Comment:
I don't thing it had `true` default by accident. Was likely implemented this way to be WAF aligned 😉

Reply:
I'd argue that default zone redundancy should be off for the following reasons:

* At the Azure Portal zone redundancy off is the default setting (you can check at the Azure portal when creating a database)
* Zone redundancy is **only** available at [selected regions](https://microsoft.sharepoint.com/:f:/r/teams/csscm/Shared%20Documents/CapacityManagementTool/[KB](https://microsoft.sharepoint.com/:f:/r/teams/csscm/Shared%20Documents/CapacityManagementTool/KB?csf=1&web=1&e=SbrEU9)?csf=1&web=1&e=SbrEU9)
* Also zone redundancy is not a security feature but a _high availability_ feature.

I agree that for a WAF deployment zone redundancy is necessary, but for "plain" deployments turning it on be by default is leading to unnecessary deployment issues (not available everywhere).

Reply:
Hey @peterbud,
I get your point, but the so-called 'reliability' pillar of WAF is one of the enforced ones for AVM. I'm pretty sure sooner or later there'll also be a failing PSRule test if we don't enable the setting. For this reason we usually aways enabled the setting for AVM's resource modules.
Lastly, the Portal is (unfortunately) not a reliable source for settings.

Anyways, let me loop @jtracey93 & @eriqua in here to collect their thoughts as well.

Reply:
Hi @peterbud, echoing what @AlexanderSehr mentioned above.

You can reference this FAQ page to know more about [What AVM means by WAF aligned](https://azure.github.io/Azure-Verified-Modules/faq/#what-does-avm-mean-by-waf-aligned)
In particular, the further note mentioning how both reliability and security are prioritized.

Let me add that your point about zone redundancy not being available for all locations is totally valid. 

However, as mentioned by the same note, defaulted values can be anyway altered via input parameters if required. 

This means zone redundancy should be explicitly disabled if using a non supporting location. Generally, the intention is to make it more difficult rather than easier to deploy a non WAF aligned resource.

@jtracey93 happy to hear your thoughts as well

Reply:
OK, no worries, then let's keep `zoneRedundant` by default on.